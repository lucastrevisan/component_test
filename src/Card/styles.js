import styled, { css } from "styled-components";
import { white, blue, white40 } from "../Common/styles/settings";
import { rem, media } from "../Common/styles/tools/index";

export const Title = styled.h1`
  color: ${blue};
  font-family: "Lato";
  font-size: ${rem(11)};
  font-weight: bold;
  line-height: 1.45;
  text-transform: uppercase;

  ${({ titleColor }) =>
    titleColor &&
    css`
      color: ${titleColor};
    `};
`;

export const IconContainer = styled.span`
  align-items: center;
  background: ${white40};
  border-radius: ${rem(4)};
  color: ${blue};
  display: inline-flex;
  height: ${rem(24)};
  justify-content: center;
  margin-right: ${rem(9)};
  width: ${rem(24)};

  ${({ iconContainerBg }) =>
    iconContainerBg &&
    css`
      background: rgba(${iconContainerBg}, 0.2);
    `};

  ${media.md(css`
    height: ${rem(28)};
    width: ${rem(28)};
    margin-right: ${rem(13)};
  `)};
`;

export const Header = styled.header`
  align-items: center;
  display: flex;
`;

export const Section = styled.section`
  background-color: ${white};
  border-radius: ${rem(2)};
  padding: ${rem(12)};
  box-shadow: 0 ${rem(1)} ${rem(12)} ${rem(6)} rgba(211, 225, 232, 0.13);

  ${media.md(css`
    padding: ${rem(20)};
  `)};

  ${({ hasChildren }) =>
    hasChildren &&
    css`
      ${Header} {
        margin-bottom: ${rem(20)};

        ${media.md(css`
          margin-bottom: ${rem(14)};
        `)};

        ${media.lg(css`
          margin-bottom: ${rem(10)};
        `)};
      }
    `};
`;
