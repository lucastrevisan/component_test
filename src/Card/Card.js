import React from "react";
import { string, node } from "prop-types";
import Icon from "../Icon";

import { Section, Title, IconContainer, Header } from "./styles";

function Card({
  title,
  icon,
  iconColor,
  titleColor,
  iconContainerBg,
  children,
  ...props
}) {
  return (
    <Section data-test={props["data-test"]} hasChildren={!!children}>
      {(icon || title) && (
        <Header>
          {icon && (
            <IconContainer iconContainerBg={iconContainerBg}>
              <Icon type={icon} color={iconColor} />
            </IconContainer>
          )}
          {title && <Title titleColor={titleColor}>{title}</Title>}
        </Header>
      )}

      {children}
    </Section>
  );
}

Card.defaultProps = {
  title: null,
  icon: null,
  iconColor: null,
  titleColor: null,
  children: null
};

Card.propTypes = {
  title: string,
  icon: string,
  iconColor: string,
  titleColor: string,
  children: node
};

export default Card;
