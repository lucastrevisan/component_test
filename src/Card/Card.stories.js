import React from "react";
import { ThemeProvider } from "styled-components";
import { storiesOf } from "@storybook/react"; // eslint-disable-line
import { withInfo } from "@storybook/addon-info"; // eslint-disable-line

import Card from "./Card";
import { defaultTheme } from "../../../styles/settings/themes";

storiesOf("common/Card", module)
  .add(
    "with title",
    withInfo()(() => (
      <ThemeProvider theme={defaultTheme}>
        <Card title="Operation sumary" />
      </ThemeProvider>
    ))
  )
  .add(
    "with icon",
    withInfo()(() => (
      <ThemeProvider theme={defaultTheme}>
        <Card icon="Cash" />
      </ThemeProvider>
    ))
  )
  .add(
    "with title and icon",
    withInfo()(() => (
      <ThemeProvider theme={defaultTheme}>
        <Card icon="Cash" title="Operation sumary" />
      </ThemeProvider>
    ))
  )
  .add(
    "with content",
    withInfo()(() => (
      <ThemeProvider theme={defaultTheme}>
        <Card icon="Cash" title="Operation sumary">
          <p>some content</p>
        </Card>
      </ThemeProvider>
    ))
  );
