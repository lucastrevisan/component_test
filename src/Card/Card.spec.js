import React from "react";
import { shallow } from "enzyme";

import Card from "./Card";

describe("Card component", () => {
  it("should match snapshot with empty content", () => {
    expect(shallow(<Card />)).toMatchSnapshot();
  });

  it("should match snapshot with title", () => {
    expect(shallow(<Card title="ok" icon="Cash" />)).toMatchSnapshot();
  });

  it("should match snapshot with icon", () => {
    expect(shallow(<Card icon="Cash" />)).toMatchSnapshot();
  });

  it("should match snapshot with title and icon", () => {
    expect(shallow(<Card title="ok" icon="Cash" />)).toMatchSnapshot();
  });

  it("should match snapshot with children", () => {
    expect(
      shallow(
        <Card title="ok" icon="Cash">
          bla
        </Card>
      )
    ).toMatchSnapshot();
  });
});
