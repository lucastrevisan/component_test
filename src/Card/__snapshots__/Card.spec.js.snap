// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot with children 1`] = `
<styled.section
  hasChildren={true}
>
  <styled.header>
    <styled.span>
      <Icon
        color={null}
        height={null}
        type="Cash"
        width={null}
      />
    </styled.span>
    <styled.h1
      titleColor={null}
    >
      ok
    </styled.h1>
  </styled.header>
  bla
</styled.section>
`;

exports[`Card component should match snapshot with empty content 1`] = `
<styled.section
  hasChildren={false}
/>
`;

exports[`Card component should match snapshot with icon 1`] = `
<styled.section
  hasChildren={false}
>
  <styled.header>
    <styled.span>
      <Icon
        color={null}
        height={null}
        type="Cash"
        width={null}
      />
    </styled.span>
  </styled.header>
</styled.section>
`;

exports[`Card component should match snapshot with title 1`] = `
<styled.section
  hasChildren={false}
>
  <styled.header>
    <styled.span>
      <Icon
        color={null}
        height={null}
        type="Cash"
        width={null}
      />
    </styled.span>
    <styled.h1
      titleColor={null}
    >
      ok
    </styled.h1>
  </styled.header>
</styled.section>
`;

exports[`Card component should match snapshot with title and icon 1`] = `
<styled.section
  hasChildren={false}
>
  <styled.header>
    <styled.span>
      <Icon
        color={null}
        height={null}
        type="Cash"
        width={null}
      />
    </styled.span>
    <styled.h1
      titleColor={null}
    >
      ok
    </styled.h1>
  </styled.header>
</styled.section>
`;
